### PR TITLE
Proposal to expose fcntl() API as part of the Swift stdlib.

### PR DESCRIPTION
**Pull request by billabt**
Previously discussed on the Swift Evolution forum.

### CODE REVIEW

File: proposals/fcntl-additions-to-stdlib.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
+# Expose fcntl() API as part of the Swift stdlib
+
+* Proposal: [SE-NNNN]
+* Author(s): [Bill Abt](https://github.com/billabt)
+* Status: **Review**
+* Review manager: TBD
+
+## Introduction
+
+At the present time Swift does not support the calling of variadic / va_list
```
Comment (gribozavr) [inline proposals/fcntl-additions-to-stdlib.md]: Swift does support `va_list` functions, it only does not support variadic ones.

Comment (billabt) [inline proposals/fcntl-additions-to-stdlib.md]: I’ll fix the proposal…  Thanks.

-Bill

**Comment by manavgabhawala** [comment]
I think this is more of an implementation detail rather than a proposal to change the language. I am sure if you made this change to the swift std lib directly and made a pull request with the changes that they would be accepted even without any talks in swift evolution. I think everyone can agree exposing C functions better to swift is a good thing so there isn't really a need to discuss it in swift evolution.

**Comment by gribozavr** [comment]
@manavgabhawala Every change to the library API need to go through the evolution process.

**Comment by billabt** [comment]
I totally agree.  There can be side effects and who knows what as a result of adding things to the library without any sort of prior discussion.  It doesn’t take long to write up a proposal nor does it do any harm if folks discuss it…

-Bill

**Comment by ddunbar** [comment]
I feel like it might be worth having an exception if the new code just adds Glibc support for the same APIs as we already support on Darwin. That is an important thing to call out on any patch to the overlays, in any case.

**Comment by billabt** [comment]
In this case, it’s a combination.  fcntl() is currently not supported in either Darwin or Glibc.  open(), open() and see_open() are already in Darwin but not Glibc.  In the PR that I did that contains the implementation (https://github.com/apple/swift/pull/196 <https://github.com/apple/swift/pull/196>) you’ll see the last three are straight ports from the Darwin implementation.  The fcntl() implementation is entirely new and includes a full battery of test cases.  So this is basically a hybrid and should probably go thru swift-evolution.  Thoughts?

**Comment by lattner** [comment]
Daniel, I agree with you - there is nothing to design/review on that case, it is simply a parity issue.  I'll bring this up with Doug even though it doesn't relate to this specific case.

**Comment by DougGregor** [comment]
Sorry for the back-and-forth here; I think this can just be handled as a fix to the Darwin and Glibc overlays, because it is working around a compiler limitation (inability to importer variadic C functions) to expose the same API one would get from POSIX.

**Comment by lattner** [comment]
Bill, I'm sorry for the delay on this - we're all still learning how to do things.

**Comment by billabt** [comment]
No problem.  I completely understand.  So am I.  Just ask Dmitri.  😃

**Pull request closed without merging**